site/js/modules/webui/course.js

```diff
@@ -35,12 +37,123 @@ function handlerCourse(path, params, context, container) {
         cards.push(Render.makeCardObject('assignment', assignment.name, link));
     }
 
+    cards.push(Render.makeCardObject(
+        "course-action",
+        "Email Users",
+        Routing.formHashPath(Routing.PATH_EMAIL, {
+            [Routing.PARAM_COURSE]: course.id,
+        }),
+    ));
+
     container.innerHTML = `
         <h2>${course.name}</h2>
         ${Render.cards(cards)}
     `;
 }
 
+function handlerEmail(path, params, context, container) {
+    let course = context.courses[params[Routing.PARAM_COURSE]];
+    const courseLink = Routing.formHashPath(Routing.PATH_COURSE, {[Routing.PARAM_COURSE]: course.id});
+
+    const titleHTML = `
+        <span>
+            <a href='${courseLink}'>${course.id}</a>
+            / email
+        </span>
+    `;
+    Routing.setTitle(course.id, titleHTML);
+
+    container.innerHTML = `
+        <div class="email-page">
+            <div class="email-content">
+                <h2>Email Users</h2>
+                <div class="description">
+                    <p>Sumbitting this form will send an email to all users enrolled in the course.</p>
+                    <p>Please separate email addresses with commas.</p>
+                </div>
+                <div class="user-input-fields secondary-color drop-shadow">
+                    <fieldset>
+                        <div class="input-field">
+                            <label for="email-to">To</label>
+                            <input type="text" id="email-to" name="to" placeholder="Recipients" required />
+                        </div>
+                        <div class="input-field">
+                            <label for="email-cc">CC</label>
+                            <input type="text" id="email-cc" name="cc" placeholder="CC" />
+                        </div>
+                        <div class="input-field">
+                            <label for="email-bcc">BCC</label>
+                            <input type="text" id="email-bcc" name="bcc" placeholder="BCC" />
+                        </div>
+                        <div class="input-field">
+                            <label for="email-subject">Subject</label>
+                            <input type="text" id="email-subject" name="subject" required />
+                        </div>
+                        <div class="input-field">
+                            <label for="email-body">Content</label>
+                            <textarea id="email-body" name="body" rows="10" required></textarea>
```

Comment:
Email's don't require a body.

Back before semi-sync communication (e.g. Slack, or Discord) were as popular (and still at some companies),
you would send emails where only the subject was important and end it with "EOM" (end of message).
For example:
```
Subject: 1pm Meeting Moved to the Clown Conference Room EOM
```